manticore/core/cpu/abstractcpu.py

```diff
@@ -462,22 +463,17 @@ def emulate(self, instruction):
         # line present.
         del emu
 
-    #Generic string representation
-    def __str__(self):
-        '''
-        Returns a string representation of cpu state
-        
-        :rtype: str
-        :return: name and current value for all the registers. 
-        '''
-        result = ""
+    def print_instruction(self):
```

Comment:
Or even log to `register_logger` directly within those functions, then you won't need to create a multi line str just to `split` it right after. If keeping as is, maybe return a list of strings instead of building a long one and splitting it later.